mpris.c

```diff
@@ -423,6 +423,24 @@ static int mpris_metadata(sd_bus *_bus, const char *_path,
 			CK(mpris_msg_append_sas_dict(reply,
 					"xesam:comment", corrected));
 		}
+		if (ti->filename) {
+			if (is_url(ti->filename)) {
+				char corrected[u_str_print_size(ti->filename)];
+				u_to_utf8(corrected, ti->filename);
+				CK(mpris_msg_append_sas_dict(reply,
+						"xesam:url", corrected));
+			}
+			else {
+				// prepend schema for local files as described in the spec
+				// https://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/#uri
+				const char prefix[] = "file://";
+				char corrected[sizeof(prefix) - 1 + u_str_print_size(ti->filename)];
+				memcpy(corrected, prefix, sizeof(prefix) - 1);
+				u_to_utf8(corrected + sizeof(prefix) - 1, ti->filename);
+				CK(mpris_msg_append_sas_dict(reply,
+						"xesam:url", corrected));
```

Comment:
I believe this needs URL escaping (except for slashes). It's not clear based on the MPRIS spec, but at least [GLib](https://gitlab.gnome.org/GNOME/glib/-/commit/39864716d6404766c325caf18c6bef722dc060cd) and [browsers](https://url.spec.whatwg.org/#file-state) will parse/strip query strings and fragments.